### DISCUSSION TITLE
EditorConfig conformance

### DISCUSSION THREAD

**Discussion by Naheulf** [sandboxie-plus/Sandboxie#2736] [Q&A Feedback]
It seem there is a lot of editorconfig non conformities in this project. I checked with https://github.com/editorconfig-checker/editorconfig-checker and it fund 275436 errors in 860 files. Most of theses errors are trailing white-space (for most files) and Wrong indentation type(spaces instead of tabs) (mainly for C code in Sandboxie/Sandboxie)

I want to fix all of them. However I don't like the idea to create one huge PR with I don't know how many changes in 860 files. Even more because I don't know how to fully test the app with all theses changes.

Instead I opt in creating five PR with almost 200 files changed per PR as follow :
PR 1/5 `Sandboxie/apps/*` (176 files)
PR 2/5 `Sandboxie/core/*` (268 files)
PR 3/5 Other files in `Sandboxie/*` (149 files)
PR 4/5 `SandboxiePlus/*`(236 files)
PR 5/5 Other files (31 files)
This way I can create one "small" PR each time I have some time. And It reduce the difficulty of debugging if something goes wrong.

What option do you prefer ?
- One Huge PR with all files
- Five Big PB with almost 200 files each
- More PR with event less files changes per PR
- No "global" PR, only fix the translation file I want to fix. (so mix editorconfig fix and translation fix in one PR)
- Just don't care about editorconfig so no need to fix any editorconfig issue unrelated to the specific line change I want to make.

**Comment by isaak654** [comment]
> Even more because I don't know how to fully test the app with all theses changes.

CI workflow (main.yml) allows to generate test builds, for example:
https://github.com/sandboxie-plus/Sandboxie/actions/runs/4521528565

> What option do you prefer ?
> 
> - One Huge PR with all files
>  - Five Big PB with almost 200 files each
>  - More PR with event less files changes per PR
> - No "global" PR, only fix the translation file I want to fix. (so mix editorconfig fix and translation fix in one PR)
> - Just don't care about editorconfig so no need to fix any editorconfig issue unrelated to the specific line change I want to make.

Sorry, I don't know. Just in case, have you tried contacting @DavidXanatos privately?

**Reply by Naheulf** [reply]
I didn't tried to ~~spam~~ contact DavidXanatos privately. I would have preferred to use only a public contact like this discussion. However, since you suggested it, I guess I'll have to send him a private message if he doesn't respond to the discussion in the next 15 days, despite the mention you made.

**Reply by DavidXanatos** [reply]
Do we care about indentations and tailing white spaces???
That sounds quite irrelevant, for simplicity I would op comparing the project before and after the changes with a tool like BeyondCompare it can ignore irrelevant changes to C/C++ files and tell you if two files are semantically identical.

**Reply by Naheulf** [reply]
This project have rules about it in the [.editorconfig](/sandboxie-plus/Sandboxie/blob/master/.editorconfig) file. So I can assume you (or other contributors) care about that.

**Reply by isaak654** [reply]
It's been a long time, but I think I created that file in the hope it might be useful someday.
If you agree, I could take the responsibility for approving and merging several pull requests (preferably more than five), as long as any testing is exclusively on you. At the same time, I cannot guarantee that there will be no indentation problems in the future.

**Reply by DavidXanatos** [reply]
Is there perhaps some VS extension which helps to keep track of this?
Also do we preffer tabs over spaces? I think sandboxies original code used spaces mostly

**Reply by DavidXanatos** [reply]
If we want to be consistent i quess spaces are better than tabs.

**Reply by Naheulf** [reply]
> If you agree, I could take the responsibility for approving and merging several pull requests (preferably more than five)

I can create up to 275436 pull requests if you want. To fix one "error" per PR 😼.
More seriously, with 860 files to updates, this mean 17 PR if I target to update 50 files per PR. Or 43 PR with "only" 20 files updated per PR. That a lot of PR with the same title "Editorconfig conformance part X/Y". But whatever I'll choose as batch size I'll follow folder structures.
By the ways, you can review file by file and so you can takes several free days to check all the files.

> At the same time, I cannot guarantee that there will be no indentation problems in the future.

I know there is a way to block merge requests if the PR doesn't respect editorconfig. But that won't work If someone (force-)push the the repo bypassing the github workflow.

> Is there perhaps some VS extension which helps to keep track of this?

According to https://editorconfig.org/#pre-installed VS (for windows) have built-in support for editorconfig but VS Code need a plugin. But there is some limitations:
- Most editors / plugins wont report you the errors.
- They will fix errors... only if you change something to the file.
- Some will fix the full file (like the notepad++ plugin) while few may only fix edited lines
- That only works if the IDE know the .editorconfig file related to the edited plugin.
- That only works if the IDE support editorconfig (with integrated support or plugin) and that support is nor disabled.

> Also do we preffer tabs over spaces?

In my humble opinion I prefer tab-indent space-align (and so tabs for indentation). That way anyone can set the indentation width to whatever they want by editing the tab width in they IDE.

> I think sandboxies original code used spaces mostly

Yes. However there is no need to stick to the preference / coding style of the original sandboxie developer.
At the same times it seem the "-plus" part of the repository manly use tab, so either ways mean a lot of changes. And I'll let the notepad++ plugin fix most of theses errors (by opening each file, creating a dummy change like adding a trailing space or fixing one of the non-conformities and save). So most of my work (and the reviewers one) will be to review the job done by the plugin.

> If we want to be consistent i quess spaces are better than tabs.

If we want to be consistent we only need to use the same (simple) rules for every files.

Edit1: Fix missing newline to separate quotes from answers.
Edit2: This link can help you with VS DavidXanatos: https://learn.microsoft.com/en-us/visualstudio/ide/create-portable-custom-editor-options?view=vs-2022#troubleshoot-editorconfig-settings

**Reply by DavidXanatos** [reply]
> At the same times it seem the "-plus" part of the repository manly use tab
>

well i just go with whatever is the defaut by defualt in VS

how many errors are there when you change the .editorconfig to use spaces?

**Reply by DavidXanatos** [reply]
@Naheulf can you please runn that tol with default indentetion set to space how much issues doe sit find then?

**Comment by Sapour** [comment]
.editorconfig is great idea for consistency, and avoiding unnecessary diffs.

Needs a bit more thought.
